### PR TITLE
Safer derivation of package name from repo name

### LINKED ISSUE #131: Safer normalization of package-name in GitHub shorthand expansion
Currently we use the repo name as package name, but some repo names can contain invalid characters for a package (like `-`).

A simple solution is to replace all the invalid characters with a `_`. As per @olafurpg proposal a `replaceAll("[^a-zA-Z0-9]", "_")` should be enough.

### PR DESCRIPTION
**Pull request by gabro**
Closes #131 and also handles the case in which a repo name starts with a digit.

See unit tests, but in short

`some-repo` => `some_repo`

`42some-repo` => `_42some_repo`

Ugly but consistent (naming is hard)

### CODE REVIEW

File: scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala [hunk 1]
```diff
@@ -20,4 +20,19 @@ class GitHubUrlRewriteSuite extends FunSuiteLike with Matchers {
         url.toString shouldBe "https://github.com/someorg/somerepo/blob/master~1/scalafix-rewrites/src/main/scala/somerepo/scalafix/Somerepo_1_2_3.scala"
     }
   }
+
+  test("it replaces invalid characters in package name with _") {
+    Conf.Str("github:someorg/some-repo/1.2.3") match {
+      case GitHubUrlRewrite(url) =>
+        url.toString shouldBe "https://github.com/someorg/some-repo/blob/master/scalafix-rewrites/src/main/scala/some_repo/scalafix/Some-repo_1_2_3.scala"
```
Comment (olafurpg) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: I just noticed that the filename can be `Some-repo_1_2_3.scala`, what do you think about using `packageName_normVersion.scala` instead?

Comment (gabro) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: Yeah, I noticed that too but I wasn't sure how to address it. Probably underscores are more consistent with the filename, so it makes sense.

Also, are you ok with keeping the capitalization? So it would be

```scala
s"${packageName.capitalize}_${normVersion}.scala"
```

Comment (olafurpg) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: I agree, I think it's a good idea to capitalize the package(name in the filename.

Comment (gabro) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: Sure, I'll do it.

Also, do you think it makes sense to make the `packageName` lowercase?

Comment (olafurpg) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: Aha, I'm not sure. For example for https://github.com/ThoughtWorksInc/Binding.scala you would get package thoughtworksinc, if the org uses uppercase letters it's usually for a reason. WDYT?

Comment (gabro) [inline scalafix-tests/src/test/scala/scalafix/tests/GitHubUrlRewriteSuite.scala]: Right. Even though, in that specific example, they use a lowercase package themselves.
I'll leave it as is for now, I guess edge cases will pop up.

Right now instead the filename is lowercased, then capitalized. Should we keep the strategy or do we revert to now lowercasing it?

**Comment by olafurpg** [comment]
Good catch! Let's lowercase it then.

**Comment by gabro** [comment]
Done! @olafurpg
